pallets/subtensor/src/utils.rs

```diff
@@ -599,3 +597,37 @@ impl<T: Config> Pallet<T> {
         SubnetOwner::<T>::iter_values().any(|owner| *address == owner)
     }
 }
+
+pub fn log2(x: I96F32) -> I96F32 {
+    let mut y = x;
+    let mut result = I96F32::from_num(0.0);
+    while y < I96F32::from_num(1.0) {
+        y *= I96F32::from_num(2.0);
+        result -= I96F32::from_num(1.0);
+    }
+    while y >= I96F32::from_num(2.0) {
+        y *= I96F32::from_num(0.5);
+        result += I96F32::from_num(1.0)
+    }
+    result
+}
+
+pub fn powf(base: I96F32, exponent: I96F32) -> I96F32 {
+    if exponent == I96F32::from_num(0.0) {
+        return I96F32::from_num(1.0);
+    } else if exponent < I96F32::from_num(0.0) {
+        return I96F32::from_num(1.0) / powf(base, -exponent);
+    }
+
+    let mut result = I96F32::from_num(1.0);
+    let mut exp = exponent.to_num::<u64>();
+    let mut b = base;
+    while exp > 0 {
+        if exp % 2 == 1 {
+            result *= b;
+        }
+        b *= b;
+        exp /= 2;
+    }
+    result
+}
```

Comment:
as const mentioned, first one is built into `I96F32` apparently, and second one you can just self-multiply in a loop since we don't have fractional exponent. In other words don't need `powf` really just need `pow` which is a lot simpler to implement